Dockerfile

```diff
@@ -1,55 +1,24 @@
-# Use the official Ubuntu image as a parent image
-FROM python:3.12-slim
+# Use Miniforge base image
+FROM condaforge/mambaforge:latest

```

Comment:
I recommend specifying the specific version instead of using *"latest"* so that the build will be consistent and won't break unexpectedly if the latest version changes in future.
For example, we could use: *"condaforge/mambaforge:24.7.1-0"* instead of *"latest"*.
By pinning the version, we can avoid potential issues with future updates that could introduce changes we haven't accounted for yet.